xarray/tests/test_namedarray.py

```diff
@@ -171,6 +167,16 @@ def test_data(random_inputs: np.ndarray[Any, Any]) -> None:
         named_array.data = np.random.random((3, 4)).astype(np.float64)
 
 
+def test_real_and_imag() -> None:
+    named_array: NamedArray[Any, Any]
+    named_array = NamedArray(["x"], np.arange(3) - 1j * np.arange(3))
+    expected_real = np.arange(3)
+    assert np.array_equal(named_array.real.data, expected_real)
+
+    expected_imag = -np.arange(3)
+    assert np.array_equal(named_array.imag.data, expected_imag)
```

Comment:
You're not testing dtypes here. They don't match. 

- `expected_imag` is an int dtype. 
- `named_array.imag.data` is a float dtype.

The test should look something like this:
```python 
def test_real_and_imag() -> None:
    expected_real: np.ndarray[Any, np.dtype[np.float64]]
    expected_real = np.arange(3, dtype=np.float64)

    expected_imag: np.ndarray[Any, np.dtype[np.float64]]
    expected_imag = -np.arange(3, dtype=np.float64)

    arr: np.ndarray[Any, np.dtype[np.complex128]]
    arr = expected_real + 1j * expected_imag

    named_array: NamedArray[Any, np.dtype[np.complex128]]
    named_array = NamedArray(["x"], arr)

    actual_real: np.ndarray[Any, np.dtype[np.float64]] = named_array.real.data
    assert np.array_equal(actual_real, expected_real)
    assert actual_real.dtype == expected_real.dtype

    actual_imag: np.ndarray[Any, np.dtype[np.float64]] = named_array.imag.data
    assert np.array_equal(actual_imag, expected_imag)
    assert actual_imag.dtype == expected_imag.dtype
```

Reply:
thank you for the review... i'm going to address these in a follow up PR.